src/index.spec.ts

```diff
@@ -86,86 +95,107 @@ describe('EppoClient E2E test', () => {
   });
 
   it('assigns subject from overrides when experiment is not enabled', () => {
-    const mockConfigRequestor = td.object<ExperimentConfigurationRequestor>();
+    const mockConfigStore = td.object<InMemoryConfigurationStore<IExperimentConfiguration>>();
     const mockPoller = td.object<IPoller>();
-    td.when(mockConfigRequestor.getConfiguration(experimentName)).thenReturn({
+    td.when(mockConfigStore.get(flagKey)).thenReturn({
       ...mockExperimentConfig,
       overrides: {
         '1b50f33aef8f681a13f623963da967ed': 'variant-2',
       },
+      typedOverrides: {
+        '1b50f33aef8f681a13f623963da967ed': 'variant-2',
+      },
     });
-    const client = new EppoClient(mockConfigRequestor, mockPoller);
-    const assignment = client.getAssignment('subject-10', experimentName);
+    const client = new EppoServerClient(mockConfigStore, mockPoller);
+    const assignment = client.getAssignment('subject-10', flagKey);
     expect(assignment).toEqual('variant-2');
   });
 
   it('returns null when experiment config is absent', () => {
-    const mockConfigRequestor = td.object<ExperimentConfigurationRequestor>();
+    const mockConfigStore = td.object<InMemoryConfigurationStore<IExperimentConfiguration>>();
     const mockPoller = td.object<IPoller>();
-    td.when(mockConfigRequestor.getConfiguration(experimentName)).thenReturn(null);
-    const client = new EppoClient(mockConfigRequestor, mockPoller);
-    const assignment = client.getAssignment('subject-10', experimentName);
+    td.when(mockConfigStore.get(flagKey)).thenReturn(null);
+    const client = new EppoServerClient(mockConfigStore, mockPoller);
+    const assignment = client.getAssignment('subject-10', flagKey);
     expect(assignment).toEqual(null);
   });
 
   it('logs variation assignment', () => {
-    const mockConfigRequestor = td.object<ExperimentConfigurationRequestor>();
+    const mockConfigStore = td.object<InMemoryConfigurationStore<IExperimentConfiguration>>();
     const mockLogger = td.object<IAssignmentLogger>();
     const mockPoller = td.object<IPoller>();
-    td.when(mockConfigRequestor.getConfiguration(experimentName)).thenReturn(mockExperimentConfig);
+    td.when(mockConfigStore.get(flagKey)).thenReturn(mockExperimentConfig);
     const subjectAttributes = { foo: 3 };
-    const client = new EppoClient(mockConfigRequestor, mockPoller, mockLogger);
-    const assignment = client.getAssignment('subject-10', experimentName, subjectAttributes);
+    const client = new EppoServerClient(mockConfigStore, mockPoller);
+    client.setLogger(mockLogger);
+    const assignment = client.getAssignment('subject-10', flagKey, subjectAttributes);
     expect(assignment).toEqual('control');
     expect(td.explain(mockLogger.logAssignment).callCount).toEqual(1);
     expect(td.explain(mockLogger.logAssignment).calls[0].args[0].subject).toEqual('subject-10');
+    expect(td.explain(mockLogger.logAssignment).calls[0].args[0].featureFlag).toEqual(flagKey);
```

Comment:
Some checks here that the attributes for logging are there: `featureFlag`, `allocation`, and `experiment`

Reply:
🙌